aten/src/ATen/native/TensorShape.cpp

```diff
@@ -181,7 +198,9 @@ Tensor repeat(const Tensor& self, IntList repeats) {
   Tensor result = self.type().tensor(target_size);
   Tensor urtensor = result.type().alias(result);
   for (int64_t i = 0; i < xtensor.dim(); ++i) {
-    urtensor = urtensor.unfold(i, xtensor.size(i), xtensor.size(i));
+    // can't unfold with step 0, so make sure step is at least 1
+    // (it doesn't matter what it is in that case, because the size is 0).
+    urtensor = urtensor.unfold(i, xtensor.size(i), std::max<int64_t>(xtensor.size(i), 1));
```

Comment:
I wonder if unfold at size 0 should accept any step. I wonder what numpy does here.

Reply:
I didn't do that because the size of dimension of the returned tensor (according to the documentation and code) is: `(sizedim - size) / step + 1`, so it's not defined if step is 0.

NumPy doesn't have an equivalent AFAIK.